lib/persistence/repositories/AccountRepository.dart

```diff
@@ -0,0 +1,3 @@
+abstract class AccountRepository {
```

Comment:
There's no naming convention for interfaces in Dart. Java has it like this, C# has it with an `I` as prefix. In Dart everything is an interface so that's even more funky.

Regardless, I have named the implementation with `Impl` (like Java convention) to distinct the implementation from the interface. I would suggest that's the way to go but we would agree on a naming convention 👌

Reply:
Personally I would favour the c# way of doing it if there is no convention in dart, simply because our backend is in c# so it might be more recognisable for anyone working on both systems. Either way works of course, as long as we settle on a convention and stick with it 😃

Reply:
If everything is an interface in Dart, then one can consider if we even need abstract classes for our Repositories. 🤔

Reply:
We probably don't strictly need it, but it might be nice to have or is there a reason you are considering removing them?